resharper/src/resharper-unity/ShaderLab/Psi/DeclaredElements/ShaderLabDeclaredElementPresenter.cs

```diff
@@ -7,19 +9,118 @@ namespace JetBrains.ReSharper.Plugins.Unity.ShaderLab.Psi.DeclaredElements
     [PsiSharedComponent]
     public class ShaderLabDeclaredElementPresenter : IDeclaredElementPresenter
     {
+        public static ShaderLabDeclaredElementPresenter Instance => PsiShared.GetComponent<ShaderLabDeclaredElementPresenter>();
+
         public string Format(DeclaredElementPresenterStyle style, IDeclaredElement element, ISubstitution substitution,
             out DeclaredElementPresenterMarking marking)
         {
             marking = new DeclaredElementPresenterMarking();
             if (!(element is IShaderLabDeclaredElement))
                 return null;
 
-            // TODO: Type (Float, Color, etc.), display name, etc.
-            marking.NameRange = new TextRange(0, element.ShortName.Length);
-            return element.ShortName;
+            var result = new StringBuilder();
+
+            if (style.ShowEntityKind != EntityKindForm.NONE)
+            {
+                var entityKind = GetEntityKind(element);
+                if (entityKind != string.Empty)
+                {
+                    if (style.ShowEntityKind == EntityKindForm.NORMAL_IN_BRACKETS)
+                        entityKind = "(" + entityKind + ")";
+                    marking.EntityKindRange = AppendString(result, entityKind);
+                    result.Append(" ");
+                }
+            }
+
+            if (style.ShowNameInQuotes)
+                result.Append("\'");
+
+            if (style.ShowType == TypeStyle.DEFAULT)
+            {
+                var typeName = GetTypeName(element);
+                if (!string.IsNullOrEmpty(typeName))
+                {
+                    marking.TypeRange = marking.ScalarTypeRange = AppendString(result, typeName);
+                    result.Append(" ");
+                }
+            }
+
+            if (style.ShowName != NameStyle.NONE)
+            {
+                marking.NameRange = AppendString(result, element.ShortName);
+                result.Append(" ");
+            }
+
+            if (style.ShowType == TypeStyle.AFTER)
+            {
+                var typeName = GetTypeName(element);
+                if (!string.IsNullOrEmpty(typeName))
+                {
+                    result.Append(": ");
+                    marking.TypeRange = marking.ScalarTypeRange = AppendString(result, typeName);
+                    result.Append(" ");
+                }
+            }
+
+            if (style.ShowNameInQuotes)
+            {
+                TrimString(result);
+                result.Append("\'");
+            }
+
+            if (style.ShowName != NameStyle.NONE && style.ShowName != NameStyle.SHORT &&
+                style.ShowName != NameStyle.SHORT_RAW)
+            {
+                AppendDisplayName(result, element);
+            }
+
+            if (style.ShowConstantValue)
+            {
```

Comment:
Is this empty code block intentional?

Reply:
Not really. I meant to implement it, and forgot :) No-one's using it yet, though. I'll fix it.